frb_example/pure_dart/rust/src/api.rs

```diff
@@ -22,37 +22,37 @@ pub fn simple_adder(a: i32, b: i32) -> Result<i32> {
 
  Newlines are preserved.
 */
-pub fn primitive_types(my_i32: i32, my_i64: i64, my_f64: f64, my_bool: bool) -> Result<i32> {
+pub fn primitive_types(my_i32: i32, my_i64: i64, my_f64: f64, my_bool: bool) -> i32 {
     println!(
         "primitive_types({}, {}, {}, {})",
         my_i32, my_i64, my_f64, my_bool
     );
-    Ok(42)
+    42
 }
 
-pub fn primitive_u32(my_u32: u32) -> Result<u32> {
+pub fn primitive_u32(my_u32: u32) -> u32 {
     println!("primitive_u32({})", my_u32);
     assert_eq!(my_u32, 0xff112233);
     let ret = 0xfe112233;
     println!("returning {}", ret);
-    Ok(ret)
+    ret
 }
 
-pub fn handle_string(s: String) -> Result<String> {
+pub fn handle_string(s: String) -> String {
     println!("handle_string({})", &s);
     let s2 = s.clone();
-    Ok(s + &s2)
+    s + &s2
 }
 
-pub fn handle_return_unit() -> Result<()> {
+pub fn handle_return_unit() -> () {
```

Comment:
Btw if you are interested, this case can also be improved: Use may write `pub fn something() {}` without `-> ()`

Reply:
Good call. Actually, this didn't work without it! But I've just pushed an extra fix to handle it.

I left it this way to specifically test the `-> ()` case, but I did remove `-> ()` from `handle_customized_struct` which also had the same situation.